plugins/Web/plugin.py

```diff
@@ -149,38 +149,50 @@ def noIgnore(self, irc, msg):
     def getTitle(self, irc, url, raiseErrors):
         size = conf.supybot.protocols.http.peekSize()
         timeout = self.registryValue('timeout')
-        (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
-                timeout=timeout)
-        try:
-            text = text.decode(utils.web.getEncoding(text) or 'utf8',
-                    'replace')
-        except UnicodeDecodeError:
-            pass
-        if minisix.PY3 and isinstance(text, bytes):
-            if raiseErrors:
-                irc.error(_('Could not guess the page\'s encoding. (Try '
-                        'installing python-charade.)'), Raise=True)
-            else:
-                return None
         try:
-            parser = Title()
-            parser.feed(text)
-        except UnicodeDecodeError:
-            # Workaround for Python 2
-            # https://github.com/ProgVal/Limnoria/issues/1359
-            parser = Title()
-            parser.feed(text.encode('utf8'))
-        parser.close()
-        title = utils.str.normalizeWhitespace(''.join(parser.data).strip())
-        if title:
-            return (target, title)
-        elif raiseErrors:
-            if len(text) < size:
-                irc.error(_('That URL appears to have no HTML title.'),
-                        Raise=True)
-            else:
-                irc.error(format(_('That URL appears to have no HTML title '
-                                 'within the first %S.'), size), Raise=True)
+            (target, text) = utils.web.getUrlTargetAndContent(url, size=size,
+                timeout=timeout)
+        except Exception as e:
+            self.log.info('Web plugin TitleSnarfer: URL <%s> raised <%s>', url, str(e))
```

Comment:
the exception should be raised if `raiseErrors`.

Reply:
Ok. Then I think it's a good idea that I add also the second case, like I did at the end of the code with 'elif not raiseErrors:' and send these to logs. This way URLs with errors can be noticed for further investigation or at least noticed with the cause titlesnarfer not returning results. Please tell me if you think its good this way.

Reply:
You could make it log all the time, but only raise exceptions if `raiseErrors`.

Reply:
I separate these cases <if raiseErrors> goes to 'irc.error' and <if not raiseErrors> goes to logs. I'm not sure about verbosity in line 157 with 'irc.error(_('That URL raised <' + str(e)) + '>',                 Raise=True)' perhaps less verbosity and just informing that 'That URL raised errors'?